src/Service/ProducerSideServiceImplTest.cpp

```diff
@@ -63,34 +74,56 @@ class FakeProducer {
   }
 
   void SendBufferedCaptureEvents(int32_t num_to_send) {
+    absl::ReaderMutexLock lock{&context_and_stream_mutex_};
     ASSERT_NE(stream_, nullptr);
     orbit_grpc_protos::ReceiveCommandsAndSendEventsRequest request;
     request.mutable_buffered_capture_events();
     for (int32_t i = 0; i < num_to_send; ++i) {
       request.mutable_buffered_capture_events()->mutable_capture_events()->Add();
     }
+
+    is_writing_mutex_.LockWhen(absl::Condition(
+        +[](bool* is_writing) { return !*is_writing; }, &is_writing_));
+    is_writing_ = true;
     bool written = stream_->Write(request);
+    is_writing_ = false;
+    is_writing_mutex_.Unlock();
```

Comment:
Why isn't a simple mutex enough? Am I missing something?

Reply:
Oh boy, you're right...

Reply:
Done.